napari/_qt/layer_controls/qt_layer_controls_container.py

```diff
@@ -131,5 +131,6 @@ def _remove(self, event):
         controls = self.widgets[layer]
         self.removeWidget(controls)
         controls.close()
+        controls.deleteLater()
```

Comment:
huh... I would have thought that this line takes care of this.  Did you find that you need this for tests to pass?

https://github.com/napari/napari/blob/330aeff2d4510f0fbf62442dfa2447f1ba5fe898/napari/_qt/layer_controls/qt_layer_controls_base.py#L38

Reply:
I found that add this line reduces the number of referring objects.

Reply:
how annoying :joy:

Reply:
hm. Maybe this should be additionally tested?